src/dmd/mtype.d

```diff
@@ -1126,6 +1126,14 @@ extern (C++) abstract class Type : RootObject
         return mod == 0;
     }
 
+    /**
+     * Indicates wether the type is usable with `extern(C++)`.
+     */
+    final bool isExternCppCompatible() const nothrow pure @nogc @safe
+    {
+        return ty != Tarray && ty != Taarray && ty != Tsarray && ty != Tdelegate;
```

Comment:
This looks like you're disallowing static array params altogether, incl. `ref` params. Passing a static array by ref can be convenient (on the C++ side, an array param is always passed by reference, and mangles to a pointer or reference IIRC). This way, people can write `foo(myArray)` instead of `foo(myArray.ptr)` on the D side.